cwms-data-api/src/main/java/cwms/cda/data/dao/project/ProjectChildrenDao.java

```diff
@@ -0,0 +1,262 @@
+            if (locsOfKind != null) {
+                for (Map.Entry<String, List<CwmsId>> entry : locsOfKind.entrySet()) {
+                    String projId = entry.getKey();
+                    List<CwmsId> locs = entry.getValue();
+                    ProjectChildren.Builder builder = builderMap.computeIfAbsent(projId, k ->
+                            new ProjectChildren.Builder()
+                                    .withProject(new CwmsId.Builder()
+                                            .withOfficeId(office)
+                                            .withName(projId)
+                                            .build()));
+                    switch (kind) {
+                        case "EMBANKMENT":
+                            builder.withEmbankments(locs);
+                            break;
+                        case "LOCK":
+                            builder.withLocks(locs);
+                            break;
+                        case "OUTLET":
+                            builder.withOutlets(locs);
+                            break;
+                        case "TURBINE":
+                            builder.withTurbines(locs);
+                            break;
+                        case "GATE":
+                            builder.withGates(locs);
+                            break;
+                        default:
+                            break;
+                    }
+
+                }
+            }
+        }
+
+        return builderMap.values().stream()
+                .map(ProjectChildren.Builder::build)
+                .collect(Collectors.toList());
+
+    }
+
+    @Nullable
+    private Map<String, List<CwmsId>> getChildrenOfKind(String office, String projLike, String kind) {
+        switch (kind) {
+
+            case "EMBANKMENT":
+                return getEmbankmentChildren(office, projLike);
+            case "TURBINE":
+                return getTurbineChildren(office, projLike);
+            case "OUTLET":
+                return getOutletChildren(office, projLike);
+            case "LOCK":
+                return getLockChildren(office, projLike);
+            case "GATE":
+                return getGateChildren(office, projLike);
+            default:
+                return null;
+        }
+
+    }
+
+    private Map<String, List<CwmsId>> getGateChildren(String office, @Nullable String projLike) {
+        Map<String, List<CwmsId>> retval = new LinkedHashMap<>();
+        AV_GATE view = AV_GATE.AV_GATE;
+        dsl.selectDistinct(view.OFFICE_ID, view.GATE_ID, view.PROJECT_ID)
+                .from(view)
+                .where(view.OFFICE_ID.eq(office))
+                .and(caseInsensitiveLikeRegexNullTrue(view.PROJECT_ID, projLike))
+                .orderBy(view.OFFICE_ID, view.PROJECT_ID, view.GATE_ID)
+                .forEach(row -> {
+                    String projId = row.get(view.PROJECT_ID);
+                    CwmsId embankment = new CwmsId.Builder()
```

Comment:
Is it possible for you to ask your co-pilot to suggest variable name corrections for your source so it can correct embankment to gate. I've seen mis-named variables elsewhere too. Just wondering if there is an easy way to correct these.

Reply:
I tried a couple prompts but nothing was satisfying.  Easier to just fix them by hand.

Reply:
fixed them.